sbol_utilities/sbol3_genbank_conversion.py

```diff
@@ -67,8 +70,104 @@ class GenBank_SBOL3_Converter:
     BIO_STRAND_REVERSE = -1
     DEFAULT_GB_REC_VERSION = 1
 
+    def __init__(self) -> None:
+        def build_component_genbank_extension(*, identity, type_uri) -> GenBank_SBOL3_Converter.Component_GenBank_Extension:
+            """A builder function to be called by the SBOL3 parser
+            when it encounters a Component in an SBOL file.
+            :param identity: identity for new component class instance to have
+            :param type_uri: type_uri for new component class instance to have
+            """
+            # `types` is required and not known at build time.
+            # Supply a missing value to the constructor, then clear
+            # the missing value before returning the built object.
+            obj = self.Component_GenBank_Extension(identity=identity, types=[sbol3.PYSBOL3_MISSING], type_uri=type_uri)
+            # Remove the placeholder value
+            obj.clear_property(sbol3.SBOL_TYPE)
+            return obj
+        # set up logging
+        log_level = logging.INFO
+        logging.getLogger().setLevel(level=log_level)
+        # Register the builder function so it can be invoked by
+        # the SBOL3 parser to build objects with a Component type URI
+        sbol3.Document.register_builder(sbol3.SBOL_COMPONENT, build_component_genbank_extension)
+
+    class Component_GenBank_Extension(sbol3.Component):
+        """Overrides the sbol3 Component class to include fields to directly read and write 
+        extraneous properties of GenBank not storeable in any SBOL3 datafield.
+        :extends: sbol3.Component class
+        """
+        GENBANK_EXTRA_PROPERTY_NS = "http://www.ncbi.nlm.nih.gov/genbank"
+        def __init__(self, identity: str, types: Optional[Union[str, Sequence[str]]], **kwargs) -> None:
+            # instantiating sbol3 component object
+            super().__init__(identity=identity, types=types, **kwargs)
+            # Setting properties for GenBank's extraneous properties not settable in any SBOL3 field.
+            self.genbank_date          = sbol3.TextProperty(self, f"{self.GENBANK_EXTRA_PROPERTY_NS}#date"    , 0, 1)
```

Comment:
No, but these properties, at the minimum, are filled by GenBank with their defaults in case one does not provide explicit information on them while creating the GenBank SequenceRecords.

I've noticed there can be more "annotations" than these default ones, but couldn't find an exhaustive list of all possible annotations anywhere for this.

Reply:
We need a way of handling an unexpected property.
I would suggest raising an exception for now.
Then we can open an exception for considering whether we want to do something more permissive.

Reply:
Done.